🛠 Tooling: dedupe `typescript-eslint` dependencies

### Tooling Report Checklist

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have pulled the latest `main` branch of the repository.
- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

#1493 introduced duplication among the installed `typescript-eslint` packages.

Running `pnpm dedupe` surfaced https://github.com/JoshuaKGoldberg/sentences-per-line/issues/866.

This issue is opened as a reminder to rerun `pnpm dedupe` once that issue is resolved 😅 
(You can assign it to me).

<details><summary>dedupe check output</summary>
<p>

```text
pnpm dedupe --check
Progress: resolved 577, reused 491, downloaded 0, added 0, done
 ERR_PNPM_DEDUPE_CHECK_ISSUES  Dedupe --check found changes to the lockfile

Packages
@es-joy/jsdoccomment@0.76.0
└── @typescript-eslint/types 8.50.1 → 8.51.0

@typescript-eslint/eslint-plugin@8.50.0(@typescript-eslint/parser@8.50.0(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3))(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3)
└── ts-api-utils 2.1.0(typescript@5.9.3) → 2.4.0(typescript@5.9.3)

@typescript-eslint/project-service@8.50.0(typescript@5.9.3)
├── @typescript-eslint/tsconfig-utils 8.50.1(typescript@5.9.3) → 8.51.0(typescript@5.9.3)
└── @typescript-eslint/types 8.50.1 → 8.51.0

@typescript-eslint/type-utils@8.50.0(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3)
└── ts-api-utils 2.1.0(typescript@5.9.3) → 2.4.0(typescript@5.9.3)

@typescript-eslint/typescript-estree@8.50.0(typescript@5.9.3)
└── ts-api-utils 2.1.0(typescript@5.9.3) → 2.4.0(typescript@5.9.3)

@vitest/eslint-plugin@1.6.1(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3)(vitest@4.0.13(@types/debug@4.1.12)(@types/node@24.10.1)(jiti@2.6.1)(yaml@2.8.1))
├── @typescript-eslint/scope-manager 8.50.1 → 8.51.0
└── @typescript-eslint/utils 8.50.1(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3) → 8.51.0(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3)

eslint-doc-generator@2.4.0(eslint@9.39.1(jiti@2.6.1))(prettier@3.7.0)(typescript@5.9.3)
└── @typescript-eslint/utils 8.50.1(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3) → 8.51.0(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3)

- @typescript-eslint/project-service@8.50.1(typescript@5.9.3)
- @typescript-eslint/scope-manager@8.50.1
- @typescript-eslint/tsconfig-utils@8.50.1(typescript@5.9.3)
- @typescript-eslint/types@8.50.1
- @typescript-eslint/typescript-estree@8.50.1(typescript@5.9.3)
- @typescript-eslint/utils@8.50.1(eslint@9.39.1(jiti@2.6.1))(typescript@5.9.3)
- @typescript-eslint/visitor-keys@8.50.1
- ts-api-utils@2.1.0(typescript@5.9.3)
```

</p>
</details> 

### Additional Info

- Blocked by https://github.com/JoshuaKGoldberg/sentences-per-line/issues/866

--- a/docs/rules/order-properties.md
+++ b/docs/rules/order-properties.md
@@ -54,7 +54,7 @@ Pass in:
 - `"legacy"` - to order properties specified by [npm documentation](https://docs.npmjs.com/cli/v10/configuring-npm/package-json).
 - `"sort-package-json"` - to order properties by the default order specified in [sort-package-json](https://github.com/keithamus/sort-package-json).
 - `Array<string>` - to specify an array of top-level package properties to lint sorting on only those properties.
-All properties not in this collection will be sorted by "sort-package-json" specifications.
+  All properties not in this collection will be sorted by "sort-package-json" specifications.
 
 ```json
 {

--- a/docs/rules/require-attribution.md
+++ b/docs/rules/require-attribution.md
@@ -14,9 +14,9 @@ When publishing a package, it's helpful to include some amount of attribution.
 The npm supports two ways of defining attribution in a `package.json`:
 
 - `author`: this is either a string with name, email, and url combined, or an object with `name`, `email`, and `url`.
-This is generally the original creator of the package, or sole maintainer in smaller projects.
+  This is generally the original creator of the package, or sole maintainer in smaller projects.
 - `contributors`: a list of all collaborators contributing to the project.
-Each item in the array has the same `name`, `email`, and `url` properties as `author` has.
+  Each item in the array has the same `name`, `email`, and `url` properties as `author` has.
 
 ### Examples with Default Options (preferContributorsOnly: false)
 

--- a/package.json
+++ b/package.json
@@ -98,7 +98,6 @@
 		"prettier": "3.7.0",
 		"prettier-plugin-curly": "0.4.0",
 		"prettier-plugin-packagejson": "2.5.19",
-		"prettier-plugin-sentences-per-line": "0.2.0",
 		"prettier-plugin-sh": "0.18.0",
 		"tsdown": "0.18.0",
 		"typescript": "5.9.3",

--- a/prettier.config.js
+++ b/prettier.config.js
@@ -7,7 +7,8 @@ const config = {
 	plugins: [
 		"prettier-plugin-curly",
 		"prettier-plugin-packagejson",
-		"prettier-plugin-sentences-per-line",
+		// To be restored once https://github.com/JoshuaKGoldberg/sentences-per-line/issues/866 is resolved
+		// "prettier-plugin-sentences-per-line",
 		"prettier-plugin-sh",
 	],
 	useTabs: true,

